### ISSUE TITLE
Runtime error when getting the composite/info or composite/health

### ISSUE THREAD

**Issue by mmalandra-kb4** [reportportal/service-index#14]
Index Version: `5.0.10`
`TRAEFIK_V2_MODE=true`

Receiving the following stack trace when attempting to get the composite/info or composite/health endpoints.

```
Panic: runtime error: invalid memory address or nil pointer dereference
goroutine 20 [running]:
runtime/debug.Stack(0x48, 0x0, 0x0)
/usr/lib/go-1.13/src/runtime/debug/stack.go:24 +0x9d
runtime/debug.PrintStack()
/usr/lib/go-1.13/src/runtime/debug/stack.go:16 +0x22
github.com/go-chi/chi/middleware.Recoverer.func1.1(0xc00016f400, 0x14e07e0, 0xc0002a22a0)
/volumeC/workspace/service-index-RELEASE/pkg/mod/github.com/go-chi/chi@v4.0.2+incompatible/middleware/recoverer.go:28 +0x1b6
panic(0x11b50c0, 0x1d63f90)
/usr/lib/go-1.13/src/runtime/panic.go:679 +0x1b2
github.com/reportportal/service-index/traefik.(*Aggregator).getNodesInfoV2(0xc000376420, 0x40c176, 0xc00036d870, 0x10)
/volumeC/workspace/service-index-RELEASE/src/github.com/reportportal/service-index/traefik/traefik.go:191 +0x32e
github.com/reportportal/service-index/traefik.(*Aggregator).aggregate(0xc000376420, 0xc00036d870, 0x11cf400)
/volumeC/workspace/service-index-RELEASE/src/github.com/reportportal/service-index/traefik/traefik.go:130 +0x4a
github.com/reportportal/service-index/traefik.(*Aggregator).AggregateInfo(0xc000376420, 0x1)
/volumeC/workspace/service-index-RELEASE/src/github.com/reportportal/service-index/traefik/traefik.go:111 +0x5c
main.main.func1.2(0x7fe0adc760c0, 0xc000388e80, 0xc00016f500)
/volumeC/workspace/service-index-RELEASE/src/github.com/reportportal/service-index/main.go:61 +0x33
net/http.HandlerFunc.ServeHTTP(0xc000376440, 0x7fe0adc760c0, 0xc000388e80, 0xc00016f500)
/usr/lib/go-1.13/src/net/http/server.go:2007 +0x44
github.com/go-chi/chi.(*Mux).routeHTTP(0xc0003018c0, 0x7fe0adc760c0, 0xc000388e80, 0xc00016f500)
/volumeC/workspace/service-index-RELEASE/pkg/mod/github.com/go-chi/chi@v4.0.2+incompatible/mux.go:425 +0x278
net/http.HandlerFunc.ServeHTTP(0xc00036d6d0, 0x7fe0adc760c0, 0xc000388e80, 0xc00016f500)
/usr/lib/go-1.13/src/net/http/server.go:2007 +0x44
github.com/go-chi/chi/middleware.RequestLogger.func1.1(0x14e07e0, 0xc0002a22a0, 0xc00016f400)
/volumeC/workspace/service-index-RELEASE/pkg/mod/github.com/go-chi/chi@v4.0.2+incompatible/middleware/logger.go:46 +0x2be
net/http.HandlerFunc.ServeHTTP(0xc0003656e0, 0x14e07e0, 0xc0002a22a0, 0xc00016f400)
/usr/lib/go-1.13/src/net/http/server.go:2007 +0x44
github.com/go-chi/chi/middleware.Recoverer.func1(0x14e07e0, 0xc0002a22a0, 0xc00016f400)
/volumeC/workspace/service-index-RELEASE/pkg/mod/github.com/go-chi/chi@v4.0.2+incompatible/middleware/recoverer.go:35 +0xbb
net/http.HandlerFunc.ServeHTTP(0xc000376460, 0x14e07e0, 0xc0002a22a0, 0xc00016f400)
2020/08/06 21:07:06 "GET http://gateway:8081/composite/info HTTP/1.1" from <IP>:40786 - 000 0B in 12.216204ms
/usr/lib/go-1.13/src/net/http/server.go:2007 +0x44
github.com/go-chi/chi.(*Mux).ServeHTTP(0xc0003018c0, 0x14e07e0, 0xc0002a22a0, 0xc00016f100)
/volumeC/workspace/service-index-RELEASE/pkg/mod/github.com/go-chi/chi@v4.0.2+incompatible/mux.go:82 +0x2b2
net/http.serverHandler.ServeHTTP(0xc0002a21c0, 0x14e07e0, 0xc0002a22a0, 0xc00016f100)
/usr/lib/go-1.13/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc0000a70e0, 0x14e3ba0, 0xc000388cc0)
/usr/lib/go-1.13/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
/usr/lib/go-1.13/src/net/http/server.go:2927 +0x38e
```

Verified I can reach http://gateway:8081/api/http/services and receive a valid response from the gateway container.
Example response here:
```
[
  {
    "loadBalancer": {
      "servers": [{ "url": "http://<SERVICE_IP>:8585" }],
      "passHostHeader": true
    },
    "status": "enabled",
    "usedBy": ["api@ecs"],
    "serverStatus": { "http://<SERVICE_IP>:8585": "UP" },
    "name": "api@ecs",
    "provider": "ecs",
    "type": "loadbalancer"
  },
  {
    "status": "enabled",
    "usedBy": ["api@internal"],
    "name": "api@internal",
    "provider": "internal"
  },
  {
    "status": "enabled",
    "usedBy": ["dashboard@internal"],
    "name": "dashboard@internal",
    "provider": "internal"
  },
  {
    "loadBalancer": {
      "servers": [{ "url": "http://<SERVICE_IP>:8080" }],
      "passHostHeader": true
    },
    "status": "enabled",
    "usedBy": ["index@ecs"],
    "serverStatus": { "http://<SERVICE_IP>:8080": "UP" },
    "name": "index@ecs",
    "provider": "ecs",
    "type": "loadbalancer"
  },
  { "status": "enabled", "name": "noop@internal", "provider": "internal" },
  {
    "loadBalancer": {
      "servers": [{ "url": "http://<SERVICE_IP>:9999" }],
      "passHostHeader": true
    },
    "status": "enabled",
    "usedBy": ["uat@ecs"],
    "serverStatus": { "http://<SERVICE_IP>:9999": "UP" },
    "name": "uat@ecs",
    "provider": "ecs",
    "type": "loadbalancer"
  },
  {
    "loadBalancer": {
      "servers": [{ "url": "http://<SERVICE_IP>:8080" }],
      "passHostHeader": true
    },
    "status": "enabled",
    "usedBy": ["ui@ecs"],
    "serverStatus": { "http://<SERVICE_IP>:8080": "UP" },
    "name": "ui@ecs",
    "provider": "ecs",
    "type": "loadbalancer"
  }
]

```

**Comment by mmalandra-kb4** [comment]
I have a fix for my issue, but I'm unable to push the changes. They are as follows:
```
diff --git a/traefik/traefik.go b/traefik/traefik.go
index fadac16..d410246 100644
--- a/traefik/traefik.go
+++ b/traefik/traefik.go
@@ -109,7 +109,7 @@ func (a *Aggregator) AggregateHealth() map[string]interface{} {
 // AggregateInfo aggregates info
 func (a *Aggregator) AggregateInfo() map[string]interface{} {
        return a.aggregate(func(info *NodeInfo) (interface{}, error) {
-               var rs map[string]interface{}
+               var rs *map[string]interface{}
                _, e := a.r.R().SetResult(&rs).Get(info.GetInfoEndpoint())
                if nil != e {
                        log.Errorf("Unable to aggregate info: %v", e)
@@ -187,8 +187,10 @@ func (a *Aggregator) getNodesInfoV2() (map[string]*NodeInfo, error) {
        nodesInfo := make(map[string]*NodeInfo, len(serviceInfo))
        for _, b := range serviceInfo {
-               backName := b.Name[:strings.LastIndex(b.Name, "@")]
-               nodesInfo[backName] = &NodeInfo{URL: b.LoadBalancer.Servers[0].URL}
+               if b.LoadBalancer != nil {
+                       backName := b.Name[:strings.LastIndex(b.Name, "@")]
+                       nodesInfo[backName] = &NodeInfo{URL: b.LoadBalancer.Servers[0].URL}
+               }
        }
        return nodesInfo, nil
```

**Comment by mmalandra-kb4** [comment]
Pull request created to resolve this issue.

**Comment by avarabyeu** [comment]
PR has been merged. Thank you for the contribution!

**Comment by malmor** [comment]
Hey @avarabyeu,
I see that the fix has been merged into master: https://github.com/reportportal/service-index/pull/24
Thanks for that 🚀

But the latest available release ([v5.0.11](https://github.com/reportportal/service-index/releases/tag/v5.0.11)) is missing this.
Is a new version including this fix going to be released soon? Otherwise we can't upgrade our traefik proxy.

Thanks 👍
